#397 Install all cmake files

uilianries opened the pull request:
- Add ConfigCatch2.cmake
- Install all CMake stuff in package folder
- Update CONAN_CMAKE_MODULE_PATH to conan package root path

close https://github.com/bincrafters/community/issues/397

/cc @Quincunx271 @acgetchell

uilianries committed:
#397 Install Catch2 cmake files

- Add ConfigCatch2.cmake
- Install all CMake stuff in package folder
- Update CONAN_CMAKE_MODULE_PATH to package root path

Signed-off-by: Uilian Ries <uilianries@gmail.com>

Quincunx271 commented:
Is there a way I can test this? I don't know how. There's no `catch2/2.3.0@bincrafters/testing`

uilianries commented:
Sorry, we don't push testing packages using CI. I just uploaded it now.

But it's not working on Windows. I need to fix this before to merge.

Quincunx271 commented:
I tried this `catch2/2.3.0@bincrafters/testing` package. It seems that it installed all the cmake files under the root installation directory rather than `lib/cmake`.

uilianries commented:
Exactly, because conan appends CMAKE_MODULE_PATH with
CONAN_CMAKE_MODULE_PATH. In other works, Conan uses the package root to
save all Cmake files.

I could move to lib/cmake as well but I would need to change cmake module
path on the recipe


On Wed, Aug 15, 2018, 6:29 PM Justin Bassett <notifications@github.com>
wrote:

> I tried this catch2/2.3.0@bincrafters/testing package. It seems that it
> installed all the cmake files under the root installation directory rather
> than lib/cmake.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bincrafters/conan-catch2/pull/3#issuecomment-413343770>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEpQHTkIal5FsCVSYtIr9NGtEWOKOsADks5uRJLRgaJpZM4V9_rQ>
> .
>

Quincunx271 commented:
Actually, just moving the files to `lib/cmake/Catch2` and touching nothing else just worked. CMake knows to look under `lib/cmake/Library` for a `LibraryConfig.cmake`

uilianries commented:
Good to know, I'm going move to `lib/cmake/Catch2` again


On Wed, Aug 15, 2018 at 6:47 PM, Justin Bassett <notifications@github.com>
wrote:

> Actually, just moving the files to lib/cmake/Catch2 and touching nothing
> else just worked. CMake knows to look under lib/cmake/Library for a
> LibraryConfig.cmake
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bincrafters/conan-catch2/pull/3#issuecomment-413348653>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEpQHbip73dpHRkVHPSNxa_RVjdT3sBUks5uRJbbgaJpZM4V9_rQ>
> .
>



-- 
Cordialmente,

Uilian Ries
<uilianries@gmail.com>

uilianries committed:
#397 Move CMake stuff to lib/cmake

- Install all cmake files in lib/cmake/Catch2
- Use Catch2 target on test_package

Signed-off-by: Uilian Ries <uilianries@gmail.com>

Quincunx271 commented:
I'm bewildered as to why the appveyor build is failing. Catch2 seems to set up the version correctly, but it seems as if `find_package` is being called as `find_package(Catch2 "" REQUIRED)` (note the explicit empty string for the version). Does conan somehow override `find_package`?

uilianries committed:
#397 Skip arch validation on Windows

- Check for Win64 before to validate void_p size
- Update test package to validate package version

Signed-off-by: Uilian Ries <uilianries@gmail.com>

uilianries commented:
No, but there is some problem to CMAKE_SIZEOF_VOID_P on Windows.

I did an workaround to skip such validation on Win64.

Quincunx271 commented:
I think we can actually skip all such validation checks. IIUC, the catch package is built externally, so on my machine I just download Catch2Config.cmake, Catch2ConfigVersion.cmake, etc. The problem with this is if my system's 32/64 bit-ness doesn't match the build server's (or wherever those files are created), CMake rejects the package.

We can just remove the check for 32/64 bit-ness altogether. It has no effect on header-only libraries, which Catch2 is.

Alternatively, we could submit a bug report on Catch2 itself and have them export a Catch2ConfigVersion.cmake without such checks (they'd have to create their own manually maintained Catch2ConfigVersion.cmake.in template file)
